### ISSUE TITLE
Automator and Program Help

### ISSUE THREAD

**Issue by burp61** [zweg25/HQ#14]
Hello, I am having an issue with the software. 
In automator, I have put the command in as shown:
I save the image on desktop as screenshot.png
Then on the shell script I enter:
cd desktop
curl -F "image=@./screenshot.png" 'https://xxxxxxxxxxxx.com/hq.php'
Automator Results show:
(
  "<br />",
  "<b>Notice</b>:  Undefined index: responses in <b>/storage/ssd4/xxxxxxxxx/public_html/hq.php</b> on line <b>80</b><br />",
  "There was an unexpected error"
)
My website shows:
Notice: Undefined index: image in /storage/ssd4/103/4972103/public_html/hq.php on line 5

Notice: Undefined index: image in /storage/ssd4/103/4972103/public_html/hq.php on line 6
There was an error uploading the file

What am I doing wrong, I have already uncommented the 100 lines. Also the image is uploaded onto the website.

**Comment by MrARM** [comment]
That means the google vision API isn't responding. the upload process is likely not the issue

**Comment by burp61** [comment]
Thank You for replying back, I fixed my google vision and can confirm that there is data going through it however I am now hit with this error:
There was an unexpected error
This is line 73 of the code and it means that there is no output coming, what could be the problem?

**Comment by alvinthomas135** [comment]
hey how did you fix the unexpected error on line 80?

**Comment by Tejindra** [comment]
@burp61 @MrARM @TiNivla  Yeah if any of you know how to fix the unexpected error please reply back because I've been having this problem for long but just don't know why.

**Comment by MrARM** [comment]
### To everyone: 
Please share your line 80 for better assistance, there was things you need to uncomment and I feel that some people will have different things for that.

### Lines 79-82
```
// GET TEXT OUTPUT
	$resp = json_decode($server_output, true);
	$text = $resp["responses"][0]["textAnnotations"][0]["description"];
	$text = strtolower($text);
```

Line 80 is something to do with handling the JSON response, this may mean you are getting an error or something similar.

### Possible solution to some
My big guess is the reason most people are having issues is because they aren't correctly setting up their google vision API access **OR** they aren't properly uploading images.

Check to make sure that you are correctly generating a valid request that will go to the vision API.
Place some echo statements around the request and verify that the image is encoded in base64 or accessible from the URL if you don't locally host the script. I personally use OSX/MacOS Server and host it locally, I removed the file upload function in favor of copying the screenshot with automator. 

If you guys are interested, I might make a fork with my revisions and a tutorial. Its getting late so I won't likely remember to do it tomorrow unless I get asked for it.

**Comment by TiNivla** [comment]
@MrARM I'm pretty sure I uploaded the image right but not sure about the google vision API.  I'll upload my line 80 probably sometime on Saturday cuz i got a busy schedule with work and school but a fork and tutorial would be greatly appreciated :)

**Comment by TiNivla** [comment]
@MrARM Hey so im not able to upload my line 80 because I ended up having to delete all the bot files, but my error looked exactly like @burp61 error.  Hows the fork going with your revisions?

**Comment by MrARM** [comment]
https://github.com/MrARM/HQ
Rewrote mostly
try it out

**Comment by Tejindra** [comment]
@MrARM I keep getting "There was an error uploading the file" when I open the local host url do you know how I can fix it
